### PR TITLE
Tidy up and update .gitignore

### PR DESCRIPTION
**Pull request by nikola-matic**
Removed superfluous ignores, added ignore policy for the entire `deps` directory, and added a leading slash `/` where necessary to indicate ignore directory that is found in the root of the repository.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -33,14 +32,10 @@ prerelease.txt
...
-docs/_static/robots.txt
-__pycache__
-docs/utils/*.pyc
-/deps/downloads/
-deps/install
-deps/cache
-cmake-build-*/
+/docs/_build
+/docs/_static/robots.txt
+/docs/utils/
```
Comment (ekpyron) [inline .gitignore]: Did you see anything that might create this? If not we can just drop it.

Comment (nikola-matic) [reply .gitignore]: Yeah, `/docs/utils/` can probably be removed; `_build` and `_static` are generated via `spinx-build` (I guess) when `docs.sh` is run, so these would have to stay.

Comment (cameel) [inline .gitignore]: I think this one should not be ignored:
```suggestion
```

Checking history, [it used to contain the lexer](https://github.com/ethereum/solidity/tree/v0.4.26/docs/utils), which is now a standalone library: [pygments-lexer-solidity](https://gitlab.com/veox/pygments-lexer-solidity).

File: .gitignore [hunk 2]
```diff
@@ -50,18 +45,15 @@ cmake-build-*/
 *~
 
 # IDE files
-.idea
-.vscode
-browse.VC.db
-CMakeLists.txt.user
+/.idea
+/.vscode
```
Comment (cameel) [inline .gitignore]:
```suggestion
/.idea/
/.vscode/
```

File: .gitignore (hunk 1 above)
Comment (cameel) [reply .gitignore]: https://github.com/ethereum/solidity/pull/13355#discussion_r940186007

**Review by cameel — APPROVED** [review]
Approving, but please do not merge until we're done with the release.

**Pull request merged**